src/sync/partial_sync.cpp

```diff
@@ -483,6 +479,42 @@ void enqueue_unregistration(Object result_set, std::function<void()> callback)
     });
 }
 
+template<typename Notifier>
+void enqueue_unregistration(Results const& result_set, std::shared_ptr<Notifier> notifier,
+                            std::function<void()> callback)
+{
+    auto realm = result_set.get_realm();
+    auto config = realm->config();
+    auto& work_queue = _impl::PartialSyncHelper::get_coordinator(*realm).partial_sync_work_queue();
+
+    // Export a reference to the __ResultSets row so we can hand it to the worker thread.
```

Comment:
This comment doesn't seem to be correct anymore? Instead, it seems we are exporting the query and re-running it.